django/http/response.py

```diff
@@ -50,6 +50,9 @@ def __init__(self, content_type=None, status=None, reason=None, charset=None):
         self.cookies = SimpleCookie()
         self.closed = False
         if status is not None:
+            if not isinstance(status, int) or status < 100 or status > 599:
+                raise ValueError('invalid HTTP status code: "{status}"'.format(
```

Comment:
Funny, I had it without quotes and worried that string versions of valid integers might get confusing e.g. "Invalid HTTP status code: 404". I think it might be better to check for TypeError separately, and possibly try to coerce any string to integer first.